v2/app/collegevine-hub/CVTemplate.js

```diff
@@ -0,0 +1,153 @@
+          // with easily readable properties like <?name?>, but we can't declare those as variables
+          // because they'll be minified.
+          // Any properly written template will NOT modify any external variables (although they can
+          // define temporary variables -- that's why we execute it inside a new scope), so the
+          // with statement should be relatively harmless.
+
+          // Oh no, it's an eval function! Why must we do this?
+          // First, it is not a security concern. Whoever supplies the HTML template already
+          // has an opportunity to run arbitrary script client-side by including <script> tags,
+          // onmouseover functions, etc., so the template provider must be trusted by the
+          // application regardless of how we do this template-filling step.
+          // Second, the template is provided as a string, which could be part of a JSON object
+          // or similar.
+          // Third, using eval (as opposed to inventing custom binding syntax) keeps the template verbosity
+          // to a minimum while giving the template writer the expressiveness to do interesting
+          // things with the data (beyond just outputting facets in their default representation).
+          // For instance, the reduce function on an array could be used to display all facet values
+          // in a variety of ways.
+          ;(function() {
+            with (item) {
+              evalResult = eval(matchString)
+            }
+          })()
+
+          // transform the result to a string, if it isn't already
+          // TODO this should take facet formatting rules into account
+          if (typeof evalResult === "number") {
+            evalResult = PivotNumber_format(evalResult)
+          } else if (evalResult instanceof Date) {
+            evalResult = evalResult.toLocaleString()
+          }
+
+          // push the string onto the result
+          outputChunks.push(evalResult)
+        } catch (e) {
+          // probably no big deal. it may have been trying to access a facet that's not set or something.
+          Seadragon2.Debug.warn(
+            "Error caught in filling template: " + e.message || e
+          )
+        }
+
+        lastIndex = curIndex
+      }
+
+      // pick up any text left after the last match
+      outputChunks.push(inputString.substring(lastIndex, inputString.length))
+
+      // put it all together in a string
+      outputString = outputChunks.join("")
+
+      if (type === "html" || type === "fakehtml") {
+        // return an HTML element whose inner HTML is based on the provided template
+        result.unsetHTML = outputString
+
+        // APP-2313: When updating the data of an existing items, Pivot never
+        // re-renders the item based on the template. We explicitly do this here:
+        var isExistingItem = result === parent
+        if (isExistingItem) {
+          result.innerHTML = outputString
+        }
```

Comment:
These are the only 4 lines I added/changed here.

Reply:
Could you please extend the comment with an explanation of how the mechanism works as well? I suspect future me might query for `APP-2313`, then need to read a whole bunch of code to understand how this actually works.

Reply:
Done 👍